datasets/swiss_judgment_prediction/swiss_judgment_prediction.py

```diff
@@ -119,56 +135,70 @@ def _split_generators(self, dl_manager):
                     data = json.loads(row)
-                    yield id_, {
-                        "id": data["id"],
-                        "year": data["year"],
-                        "text": data["text"],
-                        "label": data["label"],
-                        "language": data["language"],
-                        "region": data["region"],
-                        "canton": data["canton"],
-                        "legal area": data["legal area"],
-                    }
-        else:
+                    yield id_, self.make_dict(data)
+        elif self.config.language == "all+mt":  # original languages and mt languages
+            with open(filepath, encoding="utf-8") as f:
+                for id_, row in enumerate(f):
+                    data = json.loads(row)
+                    yield id_, self.make_dict(data)
+            last_id = id_ + 1
+            if mt_filepath:  # yield data from mt_filepath
+                with open(mt_filepath, encoding="utf-8") as f:
+                    for id_, row in enumerate(f):
+                        data = json.loads(row)
+                        yield last_id + id_, self.make_dict(data)
+        elif self.config.language in _MT_LANGUAGES:  # mt languages
+            if mt_filepath:  # yield data from mt_filepath
+                with open(mt_filepath, encoding="utf-8") as f:
+                    for id_, row in enumerate(f):
+                        data = json.loads(row)
+                        if data["language"] in self.config.language:  # "de" in "mt_de"
+                            yield id_, self.make_dict(data)
+        else:  # original languages
+            assert self.config.language in _ORIGINAL_LANGUAGES
             with open(filepath, encoding="utf-8") as f:
                 for id_, row in enumerate(f):
                     data = json.loads(row)
                     if data["language"] == self.config.language:
-                        yield id_, {
-                            "id": data["id"],
-                            "year": data["year"],
-                            "text": data["text"],
-                            "label": data["label"],
-                            "language": data["language"],
-                            "region": data["region"],
-                            "canton": data["canton"],
-                            "legal area": data["legal area"],
-                        }
+                        yield id_, self.make_dict(data)
+
+    def make_dict(self, data):
+        return {
+            "id": data["id"],
+            "year": data["year"],
+            "text": data["text"],
+            "label": data["label"],
+            "language": data["language"],
+            "region": data["region"],
+            "canton": data["canton"],
+            "legal area": data["legal area"],
+            "source_language": data["source_language"] if "source_language" in data else "n/a",
+        }
```

Comment:
This function just returns `data` with a potential additional field `source_language`.

I think this function is not needed and you just need to add one line below reading `data` in the code in `_generate_examples`:
```python
data = json.loads(row)
_ = data.setdefault("source_language", "n/a")
```